src/parser/issue.rs

```diff
@@ -1036,6 +1050,28 @@ pub(crate) fn scalar_type_cannot_be_used_in_intersection(
     .with_help("try using a different type for the intersection.")
 }
 
+pub(crate) fn literal_type_cannot_be_used_in_intersection(
+    state: &ParserState,
+    type_definition: &TypeDefinition,
+    ampersand: usize,
+) -> Issue {
+    let origin = state.source.name();
+
+    Issue::error(
+        ParserIssueCode::LiteralTypeCannotBeUsedInIntersection,
+        format!(
+            "literal type `{}` cannot be used in an intersection",
+            &type_definition,
+        ),
+        origin,
+        type_definition.initial_position(),
+        type_definition.final_position(),
+    )
+    .with_annotation(Annotation::new(origin, ampersand, ampersand + 1))
+    .with_note("a literal type is either `1`, `1.6`, `\"foo\"`, `false`, `true` or `null`.")
```

Comment:
```suggestion
    .with_note("a literal type is either a literal integer, a literal float, a literal string, `false`, `true`, or `null`.")
```

we shouldn't provide examples here, also, we use oxford comma everywhere.